### ISSUE TITLE
no-empty-character-class rule not working well with unicode flag (ES6)

### ISSUE THREAD

**Issue by dominicbarnes** [eslint/eslint#2679]
The `no-empty-character-class` rule gives a very confusing error whenever it encounters a regular expression using the ES6 `u` flag. (for unicode support)

To be clear, my regular expression does _not_ have an empty character class, and if I remove the `u` flag, it is not treated as an error anymore.

It looks like this is caused by the fact that the only recognized flags are `gimy`. (see [code](https://github.com/eslint/eslint/blob/master/lib/rules/no-empty-character-class.js#L21)) I presume adding `u` to the list would fix this problem, but not sure if that should be conditionally added with ES6 support.

This is using the latest version of ESLint. (ie: `0.22.1`)

**Comment by dominicbarnes** [comment]
Just another thought, since that rule doesn't really need to care about the flags that are used, there's no reason to deal with adding `u` conditionally. 

**tl;dr** Since the `y` flag isn't standard either, I see no reason to leave out `u`.

**Comment by dominicbarnes** [comment]
I'll submit a PR that simply adds `u` to that regex.

**Comment by nzakas** [comment]
Can you please provide:

1. The actual code your testing
2. The actual ESLint output
3. If you're using Babel-eslint?

**Comment by dominicbarnes** [comment]
Here's a distilled version:

**.eslintrc**

```js
{
  "ecmaFeatures": {
    "regexUFlag": true
  },
  "rules": {
    "no-empty-character-class": 2
  }
}
```

**test.js**

```js
var r1 = /^[^\s]+$/u;
var r2 = /^[^\s]+$/;
```

**CLI**

```
$ eslint --reset test.js

test.js
  1:9  error  Empty class  no-empty-character-class

✖ 1 problem (1 error, 0 warnings)

```

I'm expecting that both regular expressions would pass the linter, but it flags the first one as an error. (the only difference is the `u` flag)

**Comment by nzakas** [comment]
Thanks, very helpful.
